src/mappings/epochManager.ts

```diff
@@ -38,25 +38,22 @@ export function handleEpochLengthUpdate(event: EpochLengthUpdate): void {
 
   // This returns a new epoch, or current epoch, with the old epoch length
   let epoch = createOrLoadEpoch(event.block.number)
-  // We must take this epoch, update its blocks and end it, and make a new epoch
-  // Since whenever we update epoch length in the contracts we end the current epoch
-  // and create a new one
-  epoch.endBlock = event.block.number.toI32()
-  epoch.save()
+
+  // Check that the endBlock for the current epoch match what it should based on the
+  // changed epoch length
+  let newEndBlock = epoch.startBlock + event.params.epochLength.toI32()
```

Comment:
this part confuses me a bit, what are we doing exactly?

Reply:
@abarmat I think it's related to your epoch fix?

Reply:
That code is for this fix https://github.com/graphprotocol/graph-network-subgraph/pull/101

Originally the `endBlock` for the current epoch was being set to `block.number` while the real endBlock is when the epoch ends, that is startBlock + epochLength

Reply:
That we creating a shorter epoch when epoch length was changed